src/main/java/org/broadinstitute/hellbender/tools/walkers/genotyper/GenotypingEngine.java

```diff
@@ -603,34 +580,20 @@ protected final boolean passesCallThreshold(final double conf) {
         // add the MLE AC and AF annotations
         if ( alleleCountsofMLE.size() > 0 ) {
             attributes.put(GATKVCFConstants.MLE_ALLELE_COUNT_KEY, alleleCountsofMLE);
-            final ArrayList<Double> MLEfrequencies = calculateMLEAlleleFrequencies(alleleCountsofMLE, genotypes);
+            final List<Double> MLEfrequencies = calculateMLEAlleleFrequencies(alleleCountsofMLE, genotypes);
             attributes.put(GATKVCFConstants.MLE_ALLELE_FREQUENCY_KEY, MLEfrequencies);
         }
 
         if ( configuration.genotypeArgs.ANNOTATE_NUMBER_OF_ALLELES_DISCOVERED ) {
             attributes.put(GATKVCFConstants.NUMBER_OF_DISCOVERED_ALLELES_KEY, vc.getAlternateAlleles().size());
         }
 
-
         return attributes;
     }
 
-    private ArrayList<Double> calculateMLEAlleleFrequencies(final List<Integer> alleleCountsofMLE, final GenotypesContext genotypes) {
-        int AN = 0;
-        for (final Genotype g : genotypes) {
-            for (final Allele a : g.getAlleles()) {
-                if (!a.isNoCall()) {
-                    AN++;
-                }
-            }
-        }
-
-        final ArrayList<Double> MLEfrequencies = new ArrayList<>(alleleCountsofMLE.size());
-        // the MLEAC is allowed to be larger than the AN (e.g. in the case of all PLs being 0, the GT is ./. but the exact model may arbitrarily choose an AC>1)
-        for (final int AC : alleleCountsofMLE ) {
-            MLEfrequencies.add(Math.min(1.0, (double) AC / (double) AN));
-        }
-        return MLEfrequencies;
+    private List<Double> calculateMLEAlleleFrequencies(final List<Integer> alleleCountsofMLE, final GenotypesContext genotypes) {
+        final long AN = genotypes.stream().flatMap(g -> g.getAlleles().stream()).filter(Allele::isCalled).count();
+        return alleleCountsofMLE.stream().map(AC -> Math.min(1.0, (double) AC / AN)).collect(Collectors.toList());
```

Comment:
With dense streaming expressions like this I think it's good practice to include a one-line comment before each statement describing briefly what the intent is.